config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java

```diff
@@ -241,7 +241,9 @@ public ZoneEndpoint zoneEndpoint(InstanceName instance, ZoneId zone, ClusterSpec
         if (   zone.environment().isTest()
             && instances().stream()
```

Comment:
Removing this requires changes across model-config-api etc etc. That's going to be a bit risky even with flag. I can try that separately and in a later PR.

Reply:
What kind of changes are required if this is removed? Updating unit tests?

Reply:
I want to guard removing this with a flag.  The flag must either be available in this class but the fan-out seems daunting.  Or, it needs to be passed as a parameter of the zoneEndpoint() method, which crosses the config-model boundary, which means a new method must be added.  Evaluating the flag upstream will then depends on the application being on a new Vespa version. The flag needs to be evaluated in ModelContextImpl. And disabling the flag triggers only on the next redeploy (which should be OK AFAIK).